src/essence/Tools/Isochrone/IsochroneTool_Algorithm.js

```diff
@@ -0,0 +1,232 @@
+    constructor(valueArr, indexArr) {
+        this.contents = [];
+        this.valueArr = valueArr;
+        this.indexArr = indexArr;
+    }
+
+    /**
+     * Compare the cost of the pixels at indexes `i` and `j`.
+     * Return `true` if `i > j`, `false` otherwise.
+     */
+    compare(i, j) {
+        const iVal = this.contents[i];
+        const jVal = this.contents[j];
+        return getPx(this.valueArr, iVal) > getPx(this.valueArr, jVal);
+    }
+
+    /** Swap the pixels at indexes `i` and `j` */
+    swap(i, j) {
+        const iVal = this.contents[i];
+        const jVal = this.contents[j];
+        setPx(this.indexArr, iVal, j);
+        setPx(this.indexArr, jVal, i);
+        this.contents[i] = jVal;
+        this.contents[j] = iVal;
+    }
+
+    /** Move the pixel at `index` up to a correct position within the heap */
+    bubbleUp(index) {
+        let parentIndex = h_getParentIndex(index);
+        while (parentIndex >= 0 && this.compare(parentIndex, index)) {
+            this.swap(index, parentIndex);
+            index = parentIndex;
+            parentIndex = h_getParentIndex(index);
+        }
+    }
+
+    /** Move the pixel at the head of the heap down to a correct position within the heap */
+    bubbleDown() {
+        let index = 0, childIndex = 1;
+        let swapped = true;
+        const length = this.contents.length;
+        while (swapped && childIndex < length) {
+            swapped = false;
+            if (childIndex !== length - 1 && this.compare(childIndex, childIndex + 1)) {
+                childIndex++;
+            }
+            if (this.compare(index, childIndex)) {
+                this.swap(index, childIndex);
+                swapped = true;
+                index = childIndex;
+                childIndex = h_getChildIndex(index);
+            }
+        }
+    }
+
+    /** Insert a pixel coordinate tuple into the heap */
+    insert(px) {
+        let index = this.contents.length;
+        this.contents.push(px);
+        setPx(this.indexArr, px, index);
+        this.bubbleUp(index);
+    }
+
+    /** Remove the pixel coordinate tuple with the lowest cost from the heap */
+    remove() {
+        const result = this.contents[0];
+        setPx(this.indexArr, result, -1); //Mark pixel as removed
+        const lastPx = this.contents.pop();
+        if (this.contents.length > 0) {
+            this.contents[0] = lastPx;
+            setPx(this.indexArr, lastPx, 0);
+            this.bubbleDown();
+        }
+        return result;
+    }
+}
+
+/**
+ * Values to return in the backlink array. Direction back to the start ordered ascending clockwise
+ * from the right.
+ */
+const moveToBacklink = [
+    [ 0,  4,  0,  6,  0],
+    [ 2,  3,  5,  7,  8],
+    [ 0,  1,  0,  9,  0],
+    [16, 15, 13, 11, 10],
+    [ 0, 14,  0, 12,  0]
+];
+
+/**
+ * Moves from a pixel to its 8 immediate neighbors.
+ * A purely distance-based model will produce an octagon with these moves.
+ */
+const queenMoves = [
```

Comment:
So she's restricting her movement to not make the king feel inferior?

Reply:
Don't know a lot about chess haha - I believe that's a term I saw somewhere else, and I didn't think about it too much. Would probably more correctly be called `kingMoves`.

Reply:
`queen` is fine; the comment and 8 elements explain it pretty easily.

Reply:
And those moves can compose to queen moves, so it's not as if it's completely inaccurate.